improver/developer_tools/metadata_interpreter.py

```diff
@@ -0,0 +1,578 @@
+                            f"has {perc_coord.name()}"
+                        )
+
+                    if perc_coord.units != "%":
+                        self.errors.append(
+                            "Percentile coordinate should have units of %, "
+                            f"has {perc_coord.units}"
+                        )
+
+            self.check_cell_methods(cube)
+
+        # 2) Interpret model and blend information from cube attributes
+        self.check_attributes(cube.attributes)
+
+        # 3) Check whether expected coordinates are present
+        coords = get_coord_names(cube)
+        if "spot_index" in coords:
+            self.check_spot_data(cube, coords)
+
+        if self.field_type == self.ANCIL:
+            # there is no clear standard for time coordinates on static ancillaries
+            pass
+        elif (
+            cube.coords("time")
+            and len(cube.coord_dims("time")) == 2
+            and not self.blended
+        ):
+            # 2D time coordinates are only present on global day-max diagnostics that
+            # use a local time zone coordinate. These do not have a 2D forecast period.
+            expected_coords = set(LOCAL_TIME_COORDS + UNBLENDED_TIME_COORDS)
+            expected_coords.discard("forecast_period")
+            self._check_coords_present(coords, expected_coords)
+        elif self.blended:
+            self._check_coords_present(coords, BLENDED_TIME_COORDS)
+        else:
+            self._check_coords_present(coords, UNBLENDED_TIME_COORDS)
+
+        # 4) Check points are equal to upper bounds for bounded time coordinates
+        for coord in ["time", "forecast_period"]:
+            if coord in get_coord_names(cube):
+                self._check_coord_bounds(cube, coord)
+
+        # 5) Check datatypes on data and coordinates
+        try:
+            check_mandatory_standards(cube)
+        except ValueError as cause:
+            self.errors.append(str(cause))
+
+        # 6) Raise collated errors if present
+        if self.errors:
+            raise ValueError("\n".join(self.errors))
+
+
+def display_interpretation(interpreter, verbose=False):
+    """Prints metadata interpretation in human-readable form
+
+    Args:
+        interpreter (MOMetadataInterpreter):
+            Populated instance of MOMetadataInterpreter
+        verbose (bool):
+            Optional flag to include information about the source of the
+            metadata interpretation (eg name, coordinates, attributes, etc)        
+
+    Returns:
+        str:
+            Formatted string describing metadata in human-readable form
+    """
+    if interpreter.unhandled:
+        return f"{interpreter.diagnostic} is not handled by this interpreter"
+
+    def vstring(source_metadata):
+        """Format additional message for verbose output"""
+        return f"    Source: {source_metadata}"
+
+    def format_standard_cases(interpreter):
```

Comment:
The `interpreter` argument name to this subfunction is the same as the outer function. This will work fine, but may be confusing. Suggest renaming to avoid reusing the name. Another option would be to move `format_standard_cases` outside `display_interpretation` and rename it to have a private `_` prefix.

Reply:
I've made this a private function outside `display_interpretation`.